### PR TITLE
UX-241/Modified Welcome header message on dashboard

### PR DESCRIPTION
**Pull request by knnguy**
Changes made:

- If a user has a displayname, the displayname will be displayed in the welcome header 

- If a user does not have a displayname, only 'Welcome!' will be displayed

Screen Shot 2020-08-27 at 11 56 15 AM

### CODE REVIEW

File: src/app/plugins/kubernetes/components/dashboard/DashboardPage.tsx [hunk 1]
```diff
@@ -310,9 +310,18 @@ const DashboardPage = () => {
     return [hasClusters, hasAccess, hasMonitoring].findIndex((item) => !item)
   }, [hasClusters, hasMonitoring, hasAccess])
 
+  const [users, loadingUsers] = useDataLoader(mngmUserActions.list)
```
Comment (jruddell) [inline src/app/plugins/kubernetes/components/dashboard/DashboardPage.tsx:313]: When we load the session we are storing user details, does the `displayName` not get stored in the session?

Comment (knnguy) [reply src/app/plugins/kubernetes/components/dashboard/DashboardPage.tsx:313]: No, it's not stored in session. That's why I had to load all the users and get it from there

File: src/app/utils/misc.js [hunk 2]
```diff
@@ -190,16 +190,18 @@ export const getCookieValue = (name) => {
   return val ? val.pop() : ''
 }
 
-export const normalizeUsername = (name = '') => {
-  if (!name) return name
+export const normalizeUsername = (emailAddress = '') => {
```
Comment (jruddell) [inline src/app/utils/misc.js:193]: just call this `email` instead of `emailAddress`

File: src/app/utils/misc.js [hunk 3]
```diff
@@ -190,16 +190,18 @@ export const getCookieValue = (name) => {
...
+  if (!emailAddress) return emailAddress
 
   // split emails from @ sign and take the left hand side
-  const [username] = name.split('@')
-  return username
+  const [username] = emailAddress.split('@')
+
+  // split username from  '.' for emails in this format: first.last@domain.tld => [first, last]
+  const name = username.split('.')
```
Comment (jruddell) [inline src/app/utils/misc.js:200]: I'd be careful here as emails arent always reflective of a users first and last name. If there is no `displayName` the ticket calls out just rendering `Welcome` lets just do that. No need to use the `session.username` anymore.

**Pull request merged**